app/controllers/textings_controller.rb

```diff
@@ -0,0 +1,108 @@
+class TextingsController < ApplicationController
+
+  def create
+    # Make a request to Textellent API. If request successful we update our DB
+    response = post_textellent(get_db_data)
+
+    if response['status'] == 'success'
+      recipient = TextingRecipient.find_by(phone_number: texting_params[:phone_number])
+
+      if recipient
+        texting = Texting.new(
+          texting_recipient_id: recipient.id,
+          service_id: texting_params[:service_id]
+          )
+
+        if texting.save
+          render status: :ok, json: { message:'Message sent' }
+        else
+          render status: :bad_request
+        end
+
+      else
+        recipient = TextingRecipient.new(
+          recipient_name: texting_params[:recipient_name],
+          phone_number: texting_params[:phone_number]
+          )
+
+        recipient.save
+
+        texting = Texting.new(
+          texting_recipient_id: recipient.id,
+          service_id: texting_params[:service_id]
+          )
+
+        if texting.save
+          render status: :ok, json: { message:'Message sent' }
+        else
+          render status: :bad_request
+        end
+
+      end
+
+    else
+      render status: :bad_request, json: { error: response['message'] }
```

Comment:
Two comments here:

1. I don't think it's necessarily a good idea to return Textellent's error message to our web app client request. This would expose the internals of the interaction between our web app API server and Textellent's API, which could possibly contain sensitive information that we wouldn't want everyone to be able to see. Anything that is returned to the SF Service Guide web client could be visible to anyone who happens to have the Developer Tools Console open.

    We could probably just log an error locally, such as with [Sentry](https://sentry.io/welcome/), and just return a generic error to the API client. We currently use Sentry to log our errors, and we're still using their older library named Raven: https://github.com/getsentry/sentry-ruby/tree/3f37864306fba6dc0413105cf4620a11b09cda2b/sentry-raven. I _think_ you can use `Raven.capture_message()`, although we apparently don't have an example of this on our site, and I'm finding it difficult to find documentation on Raven, since Sentry is pushing people to use their new client library.

2. Could you rearrange your code into the form of an [early return](https://www.itamarweiss.com/personal/2018/02/28/return-early-pattern.html)? You can see the linked article or other articles on the web for more details, but it's often difficult to keep track of heavily nested if statements. In cases like you have, where one branch of the `if/else` is much smaller than the other, you may actually want to handle that first and return early, which allows the rest of the function be the non-error case and not be indented.

    More specifically, I would ask that you restructure your `create` method to look more like the following:

    ```rb
    response = post_textellent(get_db_data)

    if response['status'] != 'success'
      render status: :bad_request, json: { error: response['message'] }
      return
    end

    # Rest of the non-error case just goes here
    recipient = TextingRecipient.find_by(phone_number: texting_params[:phone_number])
    ...
    ```